### ISSUE TITLE
Question:  set async DNS server

### ISSUE THREAD

**Issue by calvin2021y** [warmcat/libwebsockets#2490]
Is there a way to setup DNS server for async dns resolver ?  I try some search but not able to find it.

**Comment by lws-team** [comment]
It should discover it from your platform

**Comment by calvin2021y** [comment]
My platform DNS is not reliable,  and I dont have permission to change it.

**Comment by lws-team** [comment]
There's more work to do on this topic before you can just add an api, I have plans to extend this stuff considerably.  It's not useful to, eg, just set one at context creation time.

 - handle a list of DNS servers, not just one
 - refresh the list based on netlink changes (interfaces coming up and going down) and DHCP DNS addition and removal
 - eventually, multi-server parallel queries and statistics-keeping to find out which server to favour in the near future for best perfomance

I think I have to do at the least the first point before adding an api to add and remove server entries.

**Comment by lws-team** [comment]
Please give this a try (from _temp branch)

https://libwebsockets.org/git/libwebsockets/commit?id=0c653013f35538b522d73fcade069898e3293447

**Comment by calvin2021y** [comment]
```sh
lws/lib/plat/unix/android/android-resolv.c:29:1: error: conflicting types for 'lws_plat_asyncdns_init'
lws_plat_asyncdns_init(struct lws_context *context, lws_sockaddr46 *sa46)
^
lws/lib/core-net/./private-lib-core-net.h:1494:1: note: previous declaration is here
lws_plat_asyncdns_init(struct lws_context *context, lws_async_dns_t *dns);
```

**Comment by lws-team** [comment]
I did the other platform lws_plat_asyncdns_init()  but forgot Android... I updated _temp with it

**Comment by lws-team** [comment]
This is pushed on main now, with an additional way to set a fixed server list at context time, like 8.8.8.8 etc.

Inside, it just uses the first listed DNS server for now, even though everything is changed to use a list of them.
